heartbeat/monitors/monitor.go

```diff
@@ -30,8 +30,38 @@ import (
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/monitoring"
 )
 
+var teleRegistry *monitoring.Registry = monitoring.Default.NewRegistry("heartbeat")
+var httpRegistry *monitoring.Registry = teleRegistry.NewRegistry("heartbeat.http")
```

Comment:
@urso I initially thought I could nest by writing:

```go
var teleRegistry *monitoring.Registry = monitoring.Default.NewRegistry("heartbeat")
var httpRegistry *monitoring.Registry = teleRegistry.NewRegistry("http")
```

but discovered that despite calling `NewRegistry` on an existing namespaced registry I still needed the full path. Is this by design? If so, what is the intended purpose of being able to chain `NewRegistry`? Should I refactor this to always invoke `NewRegistry` on `monitoring.Default`?

Reply:
Why do you need the full path? Where did you see things going wrong?

When calling `(*Regsitry).NewRegistry(name)`, then it adds the new registry as a child to the parent with `name` as field. This way you build a tree.
The collectors/representation is not aware of dots like `heartbeat.http`. Actually you should not use dots for names in your registry. Depending on the metrics collector in use you might get events like:

```
{
  "heartbeat": {
    "heartbeat.http": {
      ...
    }
  },
  ...
}
```